Content.Server/Cargo/Systems/CargoSystem.Orders.cs

```diff
@@ -208,16 +208,20 @@ private void OnApproveOrderMessage(EntityUid uid, CargoOrderConsoleComponent com
                 var tradePads = GetCargoPallets(trade);
                 _random.Shuffle(tradePads);
 
-                var freePads = GetFreeCargoPallets(trade, tradePads);
-
-                foreach (var pad in freePads)
+                var freePads = new List<(EntityUid Entity, CargoPalletComponent Component, TransformComponent Transform)>(GetFreeCargoPallets(trade, tradePads));
+                if (freePads.Count >= order.OrderQuantity)
                 {
```

Comment:
Just update the method to return a list as this is unnecessarily expensive.

Reply:
Yeah, that was stupid of me.